src/dialog-editor/components/modal-field-template/modalFieldTemplateComponent.ts

```diff
@@ -0,0 +1,40 @@
+import * as _ from 'lodash';
+
+/**
+ * Controller for the Dialog Editor Modal Field Template component
+ * @ngdoc controller
+ * @name ModalFieldController
+ */
+class ModalFieldController {
+  public modalData: any;
+
+}
+
+/**
+ * @memberof miqStaticAssets
+ * @ngdoc component
+ * @name dialogEditorModalFieldTemplate
+ * @description
+ *    Component contains templates for the modal for each field type
+ * @example
+ * <dialog-editor-modal-field-template ng-switch-when="DialogFieldTextBox"
+ *                                     template="text-box.html"
+ *                                     modal-data="vm.modalData">
+ * </dialog-editor-modal-field-template>
+ */
+export default class ModalFieldTemplate {
+  /*@ngInject*/
+  public template = ($element: any, $attrs: any) => {
+    return require('./' + $attrs.template);
+  };
+  public scope: boolean = true;
+  public controllerAs: string = "vm";
+  public bindings: any = {
+    modalData: '=',
+    addEntry: '=?',
```

Comment:
Good one on the optional parameter. Howerver, based on the `=` I thought they are two way databinding, is there a reason not to use callback?
```javascript
public bindings: any = {
  modalData: '=',
  addEntry: '&?',
  removeEntry: '&?',
  resolveCategories: '=?',
  categories: '=?'
}
```